### PR TITLE
fix(terraform_plan): Add provisioners to TF Plan parser

### PR DESCRIPTION
**Pull request by tsmithv11**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Add provisioners to TF Plan parser to support local-exec policies

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/plan_parser.py [hunk 1]
```diff
@@ -165,6 +166,10 @@ def _prepare_resource_block(
             resource_conf[TF_PLAN_RESOURCE_CHANGE_ACTIONS] = changes.get("change", {}).get("actions") or []
             resource_conf[TF_PLAN_RESOURCE_CHANGE_KEYS] = changes.get(TF_PLAN_RESOURCE_CHANGE_KEYS) or []
 
+        provisioners = conf.get(TF_PLAN_RESOURCE_PROVISIONERS) if conf else None
+        if provisioners:
+            resource_conf[TF_PLAN_RESOURCE_PROVISIONERS] = provisioners
```
Comment (gruebel) [inline checkov/terraform/plan_parser.py:171]: I think the problem with this approach is, that the `provisioners` in HCL looks different than in the plan file. So, a custom policy would need to cover both cases.

Comment (tsmithv11) [reply checkov/terraform/plan_parser.py]: @gruebel good callout. I've updated it to match as best I could, but note that Terraform transforms the command and removes some of it, so it won't match completely. Now this policy:

```yaml
metadata:
 name: "Test"
 id: "CKV2_CUSTOM_1"
 guidelines: "Policy description"
 category: "GENERAL_SECURITY"
scope:
  provider: "aws"
definition:
  cond_type: attribute
  resource_types:
    - aws_instance
  attribute: "provisioner/local-exec"
  operator: not_exists
```

Works for both TF and Plan files using all methods documented here: https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax

Screenshot 2023-10-19 at 1 23 08 AM
Screenshot 2023-10-19 at 1 22 32 AM

Comment (gruebel) [reply checkov/terraform/plan_parser.py]: sounds good

**Review by gruebel — APPROVED** [review]
🍻

**Review by lirshindalman — APPROVED** [review]

**Comment by junhu73** [comment]
@tsmithv11 ,  do you think there is bug here with item['expressions']['command'] since type is not checked and remote and file provisioners (or future provisioners) does not have the key "command" in the Terraform plan file.  We are seeing this error with checkov (where  checkov just exits silently and indicates nothing scanned.

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/checkov/common/parallelizer/parallel_runner.py", line 72, in func_wrapper
    result = original_func(*item)
  File "/usr/local/lib/python3.9/dist-packages/checkov/common/runners/runner_registry.py", line 835, in _parallel_run
    report = runner.run(
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_runner.py", line 118, in run
    self.definitions, definitions_raw = create_definitions(root_folder, files, runner_filter, parsing_errors)
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_utils.py", line 54, in create_definitions
    current_tf_definitions, current_definitions_raw = parse_tf_plan(file, out_parsing_errors)
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_parser.py", line 395, in parse_tf_plan
    module_blocks = _find_child_modules(
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_parser.py", line 258, in _find_child_modules
    resource_block, block_type, prepared = _prepare_resource_block(
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_parser.py", line 209, in _prepare_resource_block
    resource_conf["provisioner"] = _get_provisioner(provisioners)
  File "/usr/local/lib/python3.9/dist-packages/checkov/terraform/plan_parser.py", line 425, in _get_provisioner
    command_value = item['expressions']['command']
KeyError: 'command'

def _get_provisioner(input_data: List[Any]) -> List[Any]:
    result = []
    for item in input_data:
        key = item['type']
        command_value = item['expressions']['command']
        if not isinstance(command_value, list):
            command_value = [command_value]
        transformed_item = {key: {'command': command_value}}
        result.append(transformed_item)
    return result

**Comment by tsmithv11** [comment]
Thanks, @junhu73. Fixed here: https://github.com/bridgecrewio/checkov/pull/6606

**Pull request merged**